libs/cdk/src/lib/render-strategies/model.ts

```diff
@@ -0,0 +1,37 @@
+import { Notification, Observable } from 'rxjs';
+import { ChangeDetectorRef } from '@angular/core';
+
+export enum RxNotificationKind {
+  suspense = 'suspense',
+  next = 'next',
+  error = 'error',
+  complete = 'complete'
+}
+
+type NotificationExtract = 'value' | 'hasValue';
+
+export type RxNextNotification<T> = Pick<Notification<T>, NotificationExtract> & { kind: RxNotificationKind } & {error: boolean} & {complete: boolean};
+export type RxSuspenseNotification = Pick<Notification<unknown>, NotificationExtract> & { kind: RxNotificationKind.suspense } & {error: false} & {complete: false};
+export type RxErrorNotification = Pick<Notification<unknown>, NotificationExtract> & { kind: RxNotificationKind.error } & {error: any} & {complete: false};
+export type RxCompleteNotification = Pick<Notification<unknown>, NotificationExtract> & { kind: RxNotificationKind.complete } & {complete: boolean} & {error: false};
+export type RxNotification<T> = RxNextNotification<T> | RxSuspenseNotification | RxErrorNotification | RxCompleteNotification;
+
+export type RenderWork = <T = unknown>(
+  cdRef: ChangeDetectorRef,
+  scope?: object,
+  notification?: RxNotification<T>
```

Comment:
do we provide an operator like `rxMaterialize` ? and do we export it? Otherwise this can get confusing for people when they want to provide their own `StrategyCredentials`

Reply:
Yes, IMHO we do. 
This is used for triggers. But ATM we only use this in `rxLet`. Potentially we can also use it in `rxContext`?

Reply:
`rxMaterialize` is moved in lib root utils